packages/synapse-react-client/src/components/CreateTableViewWizard/CreateTableViewWizard.tsx

```diff
@@ -0,0 +1,359 @@
+import TableColumnSchemaForm, {
+  SubmitHandle,
+} from '../TableColumnSchemaEditor/TableColumnSchemaForm'
+import EntityViewScopeEditor from '../EntityViewScopeEditor/EntityViewScopeEditor'
+import { SetOptional } from 'type-fest'
+import SqlDefinedTableEditor from '../SqlDefinedTableEditor/SqlDefinedTableEditor'
+import { useCreateColumnModels } from '../../synapse-queries/table/useColumnModel'
+import EntityViewMaskEditor from '../EntityViewScopeEditor/EntityViewMaskEditor'
+import { DialogBase } from '../DialogBase'
+import {
+  CreateTableViewWizardStep,
+  getModalTitle,
+  getPreviousStep,
+  getStepAfterTypeSelection,
+  isLastStep,
+  maybeSetColumnIds,
+  maybeSetDefiningSQL,
+  maybeSetScopeIds,
+  maybeSetViewTypeMask,
+} from './CreateTableViewWizardUtils'
+import SubmissionViewScopeEditor from '../SubmissionViewScopeEditor/SubmissionViewScopeEditor'
+
+export type CreateTableViewWizardProps = {
+  open: boolean
+  parentId: string
+  onCancel: () => void
+  onComplete: (newEntityId: string) => void
+}
+
+/**
+ * Wizard to create a Synapse table, view, dataset, or other table type.
+ *
+ * See the README.md in this folder for a visualization of the flowchart
+ *
+ * @param props
+ * @constructor
+ */
+export default function CreateTableViewWizard(
+  props: CreateTableViewWizardProps,
+) {
+  const { open, parentId, onComplete, onCancel } = props
+  const [step, setStep] =
+    useState<CreateTableViewWizardStep>('CHOOSE_TABLE_TYPE')
+
+  const [name, setName] = useState('')
+  const [description, setDescription] = useState<string | undefined>(undefined)
+
+  const [entityType, setEntityType] = useState<EntityType | undefined>(
+    undefined,
+  )
+
+  const [entityViewScopeIds, setEntityViewScopeIds] = useState<string[]>([])
+  const [submissionViewScopeIds, setSubmissionViewScopeIds] = useState<
+    string[]
+  >([])
+  const [viewTypeMask, setViewTypeMask] = useState<number>(
+    ENTITY_VIEW_TYPE_MASK_FILE,
+  )
+  const isProjectView = viewTypeMask === ENTITY_VIEW_TYPE_MASK_PROJECT
+
+  const [columnModels, setColumnModels] = useState<
+    SetOptional<ColumnModel, 'id'>[]
+  >([])
+  const columnSchemaFormRef = useRef<SubmitHandle>(null)
+
+  const [sql, setSql] = useState('')
+
+  const { mutate: createEntity, error: createEntityError } = useCreateEntity({
+    onSuccess: entity => {
+      onComplete(entity.id!)
+    },
+  })
+
+  const { mutateAsync: createColumnModels, error: createColumnModelsError } =
+    useCreateColumnModels()
+
+  const onTableTypeSelected = useCallback(
+    (entityType: EntityType, viewTypeMask?: number) => {
+      setEntityType(entityType)
+      if (viewTypeMask != null) {
+        setViewTypeMask(viewTypeMask)
+      }
+      setStep(getStepAfterTypeSelection(entityType))
+    },
+    [],
+  )
+
+  const onViewSelected = useCallback(() => {
+    setStep('CHOOSE_VIEW_TYPE')
+    // Reset the scope IDs in case we switched from an EntityView to a SubmissionView
```

Comment:
These two were once one state variable, but since I separated them, this comment describes a step that is no longer necessary
```suggestion
```